test/tools/libtesteth/TestSuite.cpp

```diff
@@ -55,6 +61,25 @@ void removeComments(json_spirit::mValue& _obj)
 	}
 }
 
+testFileData readTestFile(fs::path _testFileName)
+{
+    testFileData testData;
+    bytes const byteContents = dev::contents(_testFileName);
+    string const s = asString(byteContents);
+    BOOST_REQUIRE_MESSAGE(s.length() > 0, "Contents of " + _testFileName.string() + " is empty.");
+
+    if (_testFileName.extension() == ".json")
+        json_spirit::read_string(s, testData.data);
+    else if (_testFileName.extension() == ".yml")
+        testData.data = test::parseYamlToJson(s);
+    else
+        BOOST_ERROR("Unknow test format!" + test::TestOutputHelper::get().testFile().string());
+
+    testData.hash = h256(sha3(json_spirit::write_string(testData.data, false)));
+    removeComments(testData.data);
```

Comment:
A bit weird that `readTestFile` returns hash of json differing from json that it returns, maybe move removing comments to the calling side where it's only needed (to `TestSuite::executeTest`)